avocado/utils/cpu.py

```diff
@@ -97,28 +97,20 @@ def get_cpu_arch():
     """
     Work out which CPU architecture we're running on
     """
+    cpu_table = [('^cpu.*(RS64|POWER3|Broadband Engine)', 'power'),
+                 ('^cpu.*POWER4', 'power4'),
+                 ('^cpu.*POWER5', 'power5'),
+                 ('^cpu.*POWER6', 'power6'),
+                 ('^cpu.*POWER7', 'power7'),
+                 ('^cpu.*POWER8', 'power8'),
+                 ('^cpu.*PPC970', 'power970'),
```

Comment:
Well yes, but I think we need better rules anyway and they'll produce if/elifs anyway. For example POWER could be optimized that any `POWER` reports the match and `\d+` (yes POWER9 is called POWER9).

Also we need to improve arm detection as on some platforms it does not report ARM so Avocado claims it's i386 and that is the last change I'd suggest which is to return "unknown" when no patterns are found as this can give incorrect results (s390 is probably also not i386...)